src/librustc_typeck/astconv.rs

```diff
@@ -321,18 +322,19 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         let mut explicit_lifetimes = Ok(());
         if !infer_lifetimes {
             if let Some(span_late) = def.has_late_bound_regions {
-                explicit_lifetimes = Err(GenericArgCountMismatch);
                 let msg = "cannot specify lifetime arguments explicitly \
                            if late bound lifetime parameters are present";
                 let note = "the late bound lifetime parameter is introduced here";
                 let span = args.args[0].span();
                 if position == GenericArgPosition::Value
                     && arg_counts.lifetimes != param_counts.lifetimes
                 {
+                    explicit_lifetimes = Err(GenericArgCountMismatch(Some(ErrorReported)));
                     let mut err = tcx.sess.struct_span_err(span, msg);
                     err.span_note(span_late, note);
                     err.emit();
```

Comment:
@Centril @estebank Can we make it so you have to emit an error to get an `ErrorReported` token, and maybe make it unconstructible except by the diagnostic reporting machinery?

Reply:
Oh, that'd be cute. :)

Reply:
Are we tracking this anywhere?

Reply:
Don't think so.

Reply:
Created https://github.com/rust-lang/rust/issues/69426.